PR-CHECKLIST.md

```diff
@@ -0,0 +1,20 @@
+# Checklist for merging PRs
+
+Before merging a PR the PR should follow these rules:
+
+* The PR does not have the label "hold"
+* There is an associated GitHub Issue linked to the PR
+  * A GitHub Issue is not required if the PR is not introducing a new feature or enhancement to Keycloak, or is resolving a bug where released versions of Keycloak are not affected
+* There is sufficient test coverage
+* Required documentation is included in the PR or an associated PR
```

Comment:
Maybe explicitly mention release notes and upgrade guide?
```suggestion
* Required documentation is included in the PR or an associated PR. Consider also Release Notes and Upgrading Guide
```

Reply:
Ended up changing this one to:
* Required documentation is included in the PR or an associated PR. Including updates to release notes and upgrade guide if applicable